src/compas_model/elements/beam.py

```diff
@@ -0,0 +1,285 @@
+from compas_model.interactions import BooleanModifier
+from compas_model.interactions import Modifier
+from compas_model.interactions import SlicerModifier
+
+
+class BeamFeature(Feature):
+    pass
+
+
+class BeamElement(Element):
+    """Class representing a beam element with a square section, constructed from WorldXY Frame.
+
+    Parameters
+    ----------
+    width : float
+        The width of the beam.
+    depth : float
+        The depth of the beam.
+    length : float
+        The length of the beam.
+    transformation : Optional[:class:`compas.geometry.Transformation`]
+        Transformation applied to the beam.
+    features : Optional[list[:class:`compas_model.features.BeamFeature`]]
+        Features of the beam.
+    name : Optional[str]
+        If no name is defined, the class name is given.
+
+    Attributes
+    ----------
+    width : float
+        The width of the beam.
+    depth : float
+        The depth of the beam.
+    length : float
+        The length of the beam.
+    is_support : bool
+        Flag indicating if the beam is a support.
+    frame_bottom : :class:`compas.geometry.Frame`
+        Main frame of the beam.
+    frame_top : :class:`compas.geometry.Frame`
+        Second frame of the beam.
+    axis : :class:`compas.geometry.Line`
+        Line axis of the beam.
+    section : :class:`compas.geometry.Polygon`
+        Section polygon of the beam.
+    polygon_bottom : :class:`compas.geometry.Polygon`
+        The bottom polygon of the beam.
+    polygon_top : :class:`compas.geometry.Polygon`
+        The top polygon of the beam.
+    transformation : :class:`compas.geometry.Transformation`
+        Transformation applied to the beam.
+    features : list[:class:`compas_model.features.BeamFeature`]
+        Features of the beam.
+    name : str
+        The name of the beam.
+    """
+
+    @property
+    def __data__(self) -> dict:
+        return {
+            "width": self.width,
+            "depth": self.depth,
+            "length": self.length,
+            "is_support": self.is_support,
+            "transformation": self.transformation,
+            "features": self._features,
+            "name": self.name,
+        }
+
+    def __init__(
+        self,
+        width: float = 0.1,
+        depth: float = 0.2,
+        length: float = 3.0,
+        is_support: bool = False,
+        transformation: Optional[Transformation] = None,
+        features: Optional[list[BeamFeature]] = None,
+        name: Optional[str] = None,
+    ) -> "BeamElement":
+        super().__init__(frame=Frame.worldXY(), transformation=transformation, features=features, name=name)
+
+        self.is_support: bool = is_support
+
+        self.width: float = width
+        self.depth: float = depth
+        self._length: float = length
+
+        self.points: list[list[float]] = [[-width * 0.5, -depth * 0.5, 0], [-width * 0.5, depth * 0.5, 0], [width * 0.5, depth * 0.5, 0], [width * 0.5, -depth * 0.5, 0]]
```

Comment:
why are these stored separately? (and if they need to be stored, then why not as a list of points?)